WebApi/WebApiTest/ControllerTest/PostControllerTest.cs

```diff
@@ -0,0 +1,279 @@
+                isLikedByUser = in_isLiked,
+                isPromoted = in_isPromoted
+            });
+
+            posts.Add(new PostDTO
+            {
+                id = in_id + 1,
+                author = in_author + " Second",
+                authorID = in_authorID + 1,
+                category = in_category + "321",
+                title = in_title + " One",
+                content = in_content + " Three",
+                likesCount = in_likesCount + 14,
+                datetime = DateTime.Now,
+                isLikedByUser = !in_isLiked,
+                isPromoted = !in_isPromoted
+            });
+
+            posts.Add(new PostDTO
+            {
+                id = in_id,
+                author = in_author + " Second2",
+                authorID = in_authorID,
+                category = in_category + "3",
+                title = in_title + " Onedsa",
+                content = in_content + " Three",
+                likesCount = in_likesCount + 4,
+                datetime = DateTime.Now,
+                isLikedByUser = in_isLiked,
+                isPromoted = !in_isPromoted
+            });
+
+            //Arrange
+            var mockService = new Mock<IPostService>();
+            mockService.Setup(x => x.GetAllOfUser(in_authorID)).Returns(posts.Where(p=>p.authorID == in_authorID).AsQueryable());
+            var mockLogger = new Mock<ILogger<PostController>>();
+            var controller = new PostController(mockLogger.Object, mockService.Object);
+
+            var expected = new List<PostDTO> { posts[0], posts[2] };
+            //Act
+            var actual = controller.GetUserPosts(in_authorID).ToList();
+
+            //Asset
+            Assert.True(expected.All(shouldItem => actual.Any(isItem => isItem == shouldItem)));
+        }
+
+        [Theory]
+        [InlineData(0, "Konrad Gaweda", 1, "RTV", "MyTitle", "Content", 41, false, false)]
+        [InlineData(1, "Jan Kowalski", 11, "RTV", "MyTitle1", "Conte nt1 ", 11, false, true)]
+        [InlineData(4, "Jan Gawęda", 1, "RTV", "MyTitle2", " Co ntent2 ", 33, true, false)]
+        [InlineData(7, "Konrad Kowalski", 32, "RTV", "MyTitle3", "Conten t3", 441, true, true)]
+
+        public void GetById_Test(int in_id, string in_author, int in_authorID, string in_category,
+            string in_title, string in_content, int in_likesCount, bool in_isLiked, bool in_isPromoted)
+        {
+            //Arrange
+            var mockService = new Mock<IPostService>();
+            mockService.Setup(x => x.GetById(0)).Returns(new PostDTO
+            {
+                id = in_id,
+                author = in_author,
+                authorID = in_authorID,
+                category = in_category,
+                title = in_title,
+                content = in_content,
+                likesCount = in_likesCount,
+                datetime = in_time,
+                isLikedByUser = in_isLiked,
+                isPromoted = in_isPromoted
+
+            });
```

Comment:
Tu jest powielony błąd który teraz sprawdziłem jest też u mnie.  Jak wywołujemy x.GetById(0) w setupie to powinno być x.GetById(id) gdzie id jest tym co chcesz normalnie przekazać. Moja wina, bo nie zauważyłem tego miejsca jak zmieniałem swój kod ze sztywno wpisanych wartości na InlineData